### PR TITLE
Adding Response Metadata to Slack Error Responses

### PR DESCRIPTION
**Pull request by matthewhall-ca**
Resolves https://github.com/jolicode/slack-php-api/issues/97

### CODE REVIEW

**Comment by pyrech** [comment]
Hello @matthewhall-ca. Great idea, thanks for your contribution! :tada:

**Review by pyrech — APPROVED** [review]

File: src/Exception/SlackErrorResponse.php [hunk 1]
```diff
@@ -15,17 +15,26 @@
 
 class SlackErrorResponse extends \Exception
 {
-    private $errorCode;
+    private string $errorCode;
 
-    public function __construct(string $errorCode, \Throwable $previous = null)
+    private ?array $responseMetadata;
```
Comment (pyrech) [inline src/Exception/SlackErrorResponse.php:20]: Sadly, typed properties can not be added yet as we still support PHP 7.2 (and they are available from PHP 7.4 only). Could you remove them?

**Review by pyrech — CHANGES_REQUESTED** [review]
Sorry I didn't see that CI builds fail for PHP < 7.4

File: src/Exception/SlackErrorResponse.php (hunk 1 above)
Comment (matthewhall-ca) [reply src/Exception/SlackErrorResponse.php]: I believe the `__construct` ones work, they were already in there when I coded this, just the `private` variable definitions at the top. I've removed those now!

**Review by pyrech — APPROVED** [review]

**Comment by pyrech** [comment]
Awesome, let's merge it! Thanks @matthewhall-ca

**Comment by pyrech** [comment]
I just released this feature as v4.3.0 :wink:

**Pull request merged**